rc-events-service/src/main/scala/com/epam/random_coffee/events/services/impl/EventServiceImpl.scala

```diff
@@ -1,37 +1,60 @@
 package com.epam.random_coffee.events.services.impl
 
-import com.epam.random_coffee.events.model.{ Event, EventId }
+import com.epam.random_coffee.events.model.{ EventId, RandomCoffeeEvent, UserId }
 import com.epam.random_coffee.events.repo.EventRepository
 import com.epam.random_coffee.events.services.EventService
 
+import java.time.Instant
+import java.time.temporal.ChronoUnit
 import java.util.UUID
 import scala.concurrent.{ ExecutionContext, Future }
 
 class EventServiceImpl(repo: EventRepository)(implicit ec: ExecutionContext) extends EventService {
 
-  override def create(eventName: String): Future[Event] = {
+  override def create(
+    name: String,
+    description: String,
+    eventDate: Instant,
+    author: UserId
+  ): Future[RandomCoffeeEvent] = {
     val id = EventId(UUID.randomUUID().toString)
-    val event = Event(id, eventName)
-    repo.save(event).map(_ => event)
+    val createdAt = Instant.now().truncatedTo(ChronoUnit.SECONDS)
+    val rcEvent = RandomCoffeeEvent(id, name, description, eventDate, createdAt, author)
+    repo.save(rcEvent).map(_ => rcEvent)
   }
 
-  override def get(id: EventId): Future[Option[Event]] = repo.get(id)
+  override def get(id: EventId): Future[Option[RandomCoffeeEvent]] = repo.get(id)
 
   override def delete(id: EventId): Future[Unit] =
     for {
       existingEvent <- repo.get(id)
-      nothing <- existingEvent.fold(notFoundError(id))(_ => Future.unit)
-      _ <- repo.delete(id)
+      _ <- existingEvent.fold(notFoundError(id))(Future.successful)
+      nothing <- repo.delete(id)
     } yield nothing
 
-  override def update(id: EventId, newEventName: String): Future[Event] =
+  override def update(
+    id: EventId,
+    newName: Option[String],
+    newDescription: Option[String],
+    newEventDate: Option[Instant]
+  ): Future[RandomCoffeeEvent] =
     for {
       existingEvent <- repo.get(id)
```

Comment:
Sorry for talking about unrelated changes, but it should be calling service method
```suggestion
      existingEvent <- get(id)
```

The reason for that is that when `get` method becomes more complicated (e.g. adds some privacy checks), we shouldn't duplicate logic everywhere.

Reply:
Thanks, I fixed it.